python/pyarrow/parquet/core.py

```diff
@@ -867,6 +867,10 @@ def _sanitize_table(table, new_schema, flavor):
     it will restore the timezone (Parquet only stores the UTC values without
     timezone), or columns with duration type will be restored from the int64
     Parquet column.
+write_page_index : bool, default False
```

Comment:
Side question: should we consider making this turned on by default at some point?

Reply:
Currently not, I found it's hard to implement page index pruning in current implementions. If we implements it, maybe we can change it to default.

Reply:
Even if it's  not used already, it would probably be beneficial to write files with the index enabled, for future use.
Is there a performance issue with enabling it?

Reply:
I guess most time there is no performance issue. But when user has extremly long string, we might write to much data.

Reply:
We are allowed to trim the min/max values, right?

Reply:
So if I understand correctly, we are currently not yet using the PageIndex when reading files (through the python APIs) for pruning pages when given a filter? 

Should we mention that in the docstring to note that you can already write a PageIndex, but it will not yet be used when reading using pyarrow?

Reply:
@jorisvandenbossche I've done that. By the way, we cannot filter using pyarrow, but `parquet-rs` and `parquet-mr` can optimize by it.